lib/rust/flexer/src/automata/dfa.rs

```diff
@@ -64,62 +64,89 @@ impl From<Vec<Vec<usize>>> for Matrix<state::Id> {
 
 
 
-// ===========
-// == Tests ==
-// ===========
+// ================
+// === Callback ===
+// ================
+
+/// The callback associated with an arbitrary state of a finite automaton.
+///
+/// It contains the rust code that is intended to be executed after encountering a
+/// [`pattern`](super::pattern::Pattern) that causes the associated state transition. This pattern
+/// is declared in [`Rule.pattern`](crate::group::rule::Rule::pattern).
+#[derive(Clone,Debug,PartialEq,Eq)]
+pub struct RuleExecutable {
+    /// A description of the priority with which the callback is constructed during codegen.
+    pub priority: usize,
+    /// The rust code that will be executed when running this callback.
+    pub name: String,
+}
+
+
+
+// =============
+// === Tests ===
+// =============
 
 #[cfg(test)]
 pub mod tests {
     use super::*;
     use crate::automata::state;
 
-    const I:usize = state::INVALID.id;
+    const I:usize = state::Identifier::INVALID.id;
```

Comment:
I know its not your code @iamrecursion, and I know I was blind earlier, but I beg you @Josef-Vonasek , never, ever use such names as one-letter vars! :)